### PR TITLE
Adding tests

### PR DESCRIPTION
**Pull request by benkeser**
This PR implements some additional checks of the raw and processed data.

The `Makefile` command for producing the processed data has been modified to the following workflow:
- run checks on raw data
- if pass, create clean data (using @youyifong as usual)
- run checks on clean data

I have also added a `make help_tests` command to show an in-words description of the tests that have been implemented.

### CODE REVIEW

**Review by thebioengineer — CHANGES_REQUESTED** [review]
Code looks good for what is there. 

One thing I could see adding is checks for no missing-ness in the cleaned code we expect imputation to be cleaning up.

**Comment by benkeser** [comment]
> One thing I could see adding is checks for no missing-ness in the cleaned code we expect imputation to be cleaning up.

Such checks are run in line 288 `data_clean/make_dat_proc.R`. @youyifong already had those checks in his script, so I just left them there. I do make a comment about them in my script so that they are described in the `make help_tests` command.

**Review by thebioengineer — APPROVED** [review]
Sounds like the change I wanted was already done - not sure how I missed that.

**Comment by youyifong** [comment]
Thanks! A few comments based off of what is not passing.

Wstratum/tps.stratum/demo.stratum is now expected to have NA's because URMforsubcohortsampling is expected to have NA's.

## missingness in weights ## isn't passing b/c the criterion for ph1 has just been updated to include EarlyendpointD57==0. Because we have double programming, this check may not add values and will increase maintenance overhead.

File: data_clean/make_raw_dat_check.R [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#-----------------------------------------------
+renv::activate()
```
Comment (nhejazi) [inline data_clean/make_raw_dat_check.R:2]:
```suggestion
renv::activate(here::here())
```

Comment (nhejazi) [reply data_clean/make_raw_dat_check.R:2]: This is also addressed in #299

File: data_clean/make_dat_proc.R [hunk 2]
```diff
@@ -331,9 +334,12 @@ if(has29) {
...
     }
     
-    stopifnot(
-      all(table(dat.tmp.impute$Wstratum, complete.cases(dat.tmp.impute[imp.markers])))
-    )
+    # missing markers imputed properly in each stratum?
+    for(w in unique(dat.tmp.impute$Wstratum)){
+      stopifnot(
+        all(complete.cases(dat.tmp.impute[dat.tmp.impute$Wstratum == w, imp.markers]))
+      )
```
Comment (nhejazi) [inline data_clean/make_dat_proc.R:341]:
```suggestion
      assertthat::assert_that(
        all(complete.cases(dat.tmp.impute[dat.tmp.impute$Wstratum == w, imp.markers])),
        msg = "missing markers imputed properly in each stratum?"
      )  
```

**Review by nhejazi — APPROVED** [review]
LGTM but suggested a few minor changes throughout.

**Pull request merged**